driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -475,12 +475,34 @@ static __always_inline unsigned long extract__vm_swap(struct mm_struct *mm)
  */
 static __always_inline u32 exctract__tty(struct task_struct *task)
 {
-	int index;
-	int major;
-	int minor_start;
-	READ_TASK_FIELD_INTO(&index, task, signal, tty, index);
-	READ_TASK_FIELD_INTO(&major, task, signal, tty, driver, major);
-	READ_TASK_FIELD_INTO(&minor_start, task, signal, tty, driver, minor_start);
```

Comment:
To be honest, this is very strange :thinking: 
Let's say if this solves the issue in this specific case ok, but here we are losing the advantages of having a modern BPF probe...

In kernels >= 5.11 this :point_down: 

```c 
READ_TASK_FIELD_INTO(&index, task, signal, tty, index);
READ_TASK_FIELD_INTO(&major, task, signal, tty, driver, major);
READ_TASK_FIELD_INTO(&minor_start, task, signal, tty, driver, minor_start);
```

is equal to

```c
index = task->signal->tty->index
major = task->signal->tty->driver->major
minor_start = task->signal->tty->driver->minor_start
```

So no bpf helpers at all, only direct access in memory.

While in the new approach we use `6` bpf helpers, so it is quite different in terms of perf :/ 

Have you tried a solution like this?

```c
index = READ_TASK_FIELD(task, signal, tty, index);
major = READ_TASK_FIELD(task, signal, tty, driver, major);
minor_start = READ_TASK_FIELD(task, signal, tty, driver, minor_start);
```

Reply:
if the last approach doesn't work I would add at least a comment otherwise at the next iteration we will change again the code to the initial state :rofl:

Reply:
🤷‍♀️ still same issues here also with your suggestion. Adding a comment is an excellent idea, ty.